File: robocode-tankroyale-bot-api-csharp/events/GameEndedEvent.cs
using System.Collections.Generic;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when game has just ended.
  /// </summary>
  public sealed class GameEndedEvent : IMessage
  {
    /// <summary>Number of rounds played.</summary>
    int NumberOfRounds { get; }

    /// <summary>Results of the battle.</summary>
    List<BotResults> Results { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="victimId">ID of the bot that has died.</param>
    public GameEndedEvent(int numberOfRounds, List<BotResults> results) : base() =>
      (NumberOfRounds, Results) = (numberOfRounds, results);
  }
}

File: bot-api/typescript/src/events/GameEndedEvent.ts
import { IEvent } from "./IEvent.js";
import { BotResults } from "../BotResults.js";

/** Event occurring when a game has ended. */
export class GameEndedEvent implements IEvent {
  readonly numberOfRounds: number;
  readonly results: BotResults;

  constructor(numberOfRounds: number, results: BotResults) {
    this.numberOfRounds = numberOfRounds;
    this.results = results;
  }
}
